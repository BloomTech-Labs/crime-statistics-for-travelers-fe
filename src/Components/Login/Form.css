html, body, #app, #app>div {
  height: 100%
}

.col-container {
  display: flex;
  width: 100%;
}
.col {
  flex: 1;
  padding: 16px;
}