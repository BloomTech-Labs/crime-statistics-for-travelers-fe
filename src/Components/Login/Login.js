import React from "react";
import { withFormik, Form, Field } from "formik";
import * as Yup from "yup";
import {Link} from 'react-router-dom'
import axios from "axios";
import { Redirect } from "react-router-dom";
import Swal from "sweetalert2";
import styled from "styled-components";
import "../../App.css"
// import posed from "react-pose";

const H5 = styled.h5`
  color: #b22222;
`;

const FormContainer = styled.div`
  

`;
const BTN =styled.button`
 
`
const Btn=styled(Link)``

const Label = styled.label`
  color: #f3e367;
  font-size: 2.5em;
`;

function Login({ touched, errors }) {
  const token = localStorage.getItem("token");

  if (token) {
    return <Redirect to="/" />;
  }

  return (
    <Form className="form">
      <FormContainer>
   
        <Field
          id="email"
          type="email"
          autoComplete="off"
          placeholder="email"
          name="email"
        />
        <H5>{touched.username && errors.username}</H5>
        <Label htmlFor="password">Password</Label>
        <Field
          id="password"
          type="password"
          autoComplete="off"
          placeholder="password"
          name="password"
        />
        <H5>{touched.password && errors.password}</H5> 
        <BTN className="formBTN" type="submit">
          <Btn to = "/">Login</Btn>
        </BTN>
      </FormContainer>
    </Form>
  );
}

export default withFormik({
  mapPropsToValues() {
    return {
      email: "",
      password: ""
    };
  },
  validationSchema: Yup.object().shape({
    email: Yup.string()
      .min(3, "Must be 3 characters or more")
      .max(20, "Must be less than 20 characters")
      .required("This field is required"),
    password: Yup.string()
      .min(2, "Must be 3 characters or more")
      .max(100, "Must be less than 100 characters")
      .required("Enter a password to continue")
    // passwordConfirmation: Yup.string()
    // .oneOf([Yup.ref('password'), null], 'Passwords must match'),
  }),
  handleSubmit(values, formikBag) {
    console.log(values,"Login values")
    axios
      .post(`https://usemytechstuff.herokuapp.com/api/auth/login/`, values)
      .then((response) => {
        localStorage.setItem('token', response.data.payload);
        console.log('does token data exist:', response.data.payload)
        formikBag.props.history.push('/');
        formikBag.props.setToken(response.data.payload)
      })
      .catch((e) => {
        // console.log(e.response.data && response.data);
      });
    //   {if(token === null){
    //     Swal.fire({
    //         position: 'center',
    //         type: 'error',
    //         title: 'Try Again!',
    //         showConfirmButton: false,
    //         timer: 2500
    //       })
    //   }else{
        Swal.fire({
            position: 'center',
            type: 'success',
            title: 'Welcome to your next Adventure',
            showConfirmButton: false,
            timer: 2500
          })
    //   }}

  }

})(Login);