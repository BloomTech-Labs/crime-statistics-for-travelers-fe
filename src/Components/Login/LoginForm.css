html, body, #root, #app>div {
  height: 100%
}

.col-container {
  width: 100%;
}

.col4{
  z-index: 0;
  position: absolute;
  width: 30%;
  top: 0%;
  right: 0%;
  padding: 16px;
  height: 93%;
  text-align: center;
  background-color: #3E6385;
  margin-top: 60px;
}

.col3{
  margin-top: 60px;
  z-index: 1;
  position: absolute;
  left: 0%;
  top: 0%;
  padding: 16px;
  height: 93%;
  width: 70%;
  text-align: center;
  background-color: #d6d3d3;
  box-shadow: 10px 0px 10px -8px #363636;

}
.heading-container{
  position: relative;
  top:35%;
  width: 70%;
  margin: 0 auto;
}

.auth-links{
  width: 60%;
  margin: 0 auto;
  display: flex;
}

.auth-link{
  width: 49%;
  padding: 1rem;
}

.heading{
  font-size: 2rem;
  padding: 20px;
}

.form-container{
  position: relative;
  top:20%;
  width: 60%;
  margin: 0 auto;
  background-color: #ffffff

}

input:focus {
  border-color: #007eff;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075),
    0 0 0 3px rgba(0, 126, 255, 0.1);
  outline: none;
}

input, input[palceholder] {
 text-align: center;
}

.form-input{
  height: 40px;
  margin: 5px auto;
  width: 75%;
  border: none;
  border:solid 1px #ccc;
  border-radius: 10px;
}

.login-form{
  display: flex;
  flex-direction: column;
  align-items: center;
}