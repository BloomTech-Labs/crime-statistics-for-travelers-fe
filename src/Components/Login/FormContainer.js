import React from 'react'
import styled from 'styled-components';
import './Form.css'

const PageContainer = styled.div`
  display: flex;
`;

const Left = styled.div`
  display: flex;
  flex-direction: column;
  width: 30%;
  height: 100%;
  background-color: #hhh;
`;

const Right = styled.div`
  width: 69%;
  height: 100%; 
`;


const FormContainer = () => {
  return (
    <PageContainer>
      <Left>Left</Left>
      <Right>Right</Right>
    </PageContainer>
  )
}

export default FormContainer