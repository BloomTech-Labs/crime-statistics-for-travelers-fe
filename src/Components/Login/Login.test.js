import React from "react";
import Login from "./Login";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

it("Snapshot test for Login page", () => {
  const tree = renderer
    .create(
      <Router>
        <Login />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
