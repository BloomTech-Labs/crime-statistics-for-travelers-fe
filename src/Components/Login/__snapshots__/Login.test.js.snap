// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Login page 1`] = `
<div
  className="col-container"
>
  <div
    className="col3"
  >
    <div
      className="form-container css-192m1qh"
    >
      <h1
        className="heading"
      >
        Log in to your account.
      </h1>
      <div
        className="auth-links"
      >
        <div
          className="auth-link"
        >
          <img
            alt="auth-0 link"
            src="btn_google_signin_light_normal_web.png"
          />
        </div>
        <div
          className="auth-link"
        >
          <img
            alt="Facebook auth0 button"
            src="login-facebook.png"
          />
        </div>
      </div>
      <form
        className="login-form"
        onSubmit={[Function]}
      >
        <input
          className="form-input"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
        <input
          className="form-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          className="css-17ddmg5"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </form>
    </div>
  </div>
  <div
    className="col4"
  >
    <div
      className="heading-container"
    >
      <h1
        className="css-1kdk8in"
      >
        New around here?
      </h1>
      <strong>
        <p>
          Please Create a 
        </p>
        <p>
          new user account.
        </p>
      </strong>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="css-3n2czx"
        >
          Sign Up
        </button>
      </a>
    </div>
  </div>
</div>
`;
